components/camel-salesforce/camel-salesforce-maven-plugin/src/main/resources/sobject-pojo.vm

```diff
@@ -127,24 +127,29 @@ public class $desc.Name extends AbstractDescribedSObjectBase {
         this.$propertyName = $propertyName;
     }
 
-#if ( $utility.isLookup($field) )## IS LOOKUP
+#if ( $utility.isLookup($field) )## EXTERNAL ID
 #foreach ( $reference in $field.getReferenceTo() )## REFERENCE LOOP
-#foreach ( $externalIds in $utility.externalIdsOf($reference) )## EXTERNAL ID LOOP
-#set( $relationshipAsFieldName = $field.getRelationshipName() )
+#if ( $utility.hasExternalIds($reference) )## IS REFERENCE CONTAIN EXTERNAL ID
+#set( $fieldName = $field.getName() )
+#if ( $fieldName.endsWith("Id") )## JSON PROPERTY NAME SETTER
+#set( $jsonPropertyName = $fieldName.substring(0, $fieldName.length() - 2) )
```

Comment:
What happens here if the property is `"Id"`?

Reply:
Then it would never be considered for this block of code - Id is a standard salesforce field and it never is a Lookup.